src/components/VideoSeriesPageController.tsx

```diff
@@ -0,0 +1,77 @@
+import * as React from "react";
+import { PageHero } from "./PageHero/PageHero";
+import { selectVideoSeriesPageData } from "../selectors/selectVideoSeriesPageData";
+import { VideoSeriesPage } from "./VideoSeriesPage";
+import { VideoCard } from "./VideoCard";
+import { PageHeroStencil } from "./PageHero/PageHeroStencil";
+import { useFetchData } from "../hooks/useFetchData";
+import { VideoCardStencil } from "./VideoCardStencil";
+
+const VideoSeriesPageController: React.FC = () => {
+  const search = window.location.search;
+  const params = new URLSearchParams(search);
+
+  const isDebugMode = params.get("debug") === "true";
+
+  const [hasVideoError] = React.useState(isDebugMode);
+
+  const { data, loading } = useFetchData();
+
+  const { videos, title, description } = selectVideoSeriesPageData(data);
+
+  const [selectedVideoId, setSelectedVideoId] = React.useState<string | null>(
+    null
+  );
+
+  const handleClickVideo = React.useCallback(
+    (videoId: string) => {
+      if (videoId === selectedVideoId) {
+        setSelectedVideoId(null);
+        return;
+      }
+      setSelectedVideoId(videoId);
+    },
+    [selectedVideoId]
+  );
+
+  const videosComponents = React.useMemo(() => {
+    return videos.map((video) => (
+      <VideoCard
+        key={video.id}
+        video={video}
+        onClickCta={() => handleClickVideo(video.youtubeId)}
+      />
+    ));
+  }, [handleClickVideo, videos]);
+
```

Comment:
Since I'm not checking here whether there's errors returning from fetching the initial data on load, I should at least have an error boundary.

Here's Dan Abramov's codepen https://codepen.io/gaearon/pen/wqvxGa?editors=0010